[Atharva Tyagi] iP

ty4g1 opened the pull request:
### Mahesh Dall-E

> "Dum-Dum, Mamta Dum-Dum." - Mahesh Dalle

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/ty4g1/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ty4g1 committed:
modified gitignore to ignore .class files

ty4g1 committed:
renamed bot to "Mahesh Dall-E"

ty4g1 committed:
Renamed file and implemented skeleton

ty4g1 committed:
implemented echo functionality

ty4g1 committed:
implemented add and list functionalities

ty4g1 committed:
added mark and unmark functionality

ty4g1 committed:
created and implemented Task class

ty4g1 committed:
implemented deadline, event and todo task types

ty4g1 committed:
updated add and print messages

ty4g1 committed:
fixed incorrect print bug and added text ui test

ty4g1 committed:
added error handling

ty4g1 committed:
changed to use ArrayList for task list

ty4g1 committed:
fixed mark/unmark exception handling

ty4g1 committed:
Implemented delete functionality

ty4g1 committed:
updated test files

ty4g1 committed:
Moved static methods into their respective classes

ty4g1 committed:
added enums to handle commands and switch-case

ty4g1 committed:
added javadocs

ty4g1 committed:
made Task class abstract

ty4g1 committed:
Implement save functionality

ty4g1 committed:
Revert "Implement save functionality"

This reverts commit 51e56fb152905ce16917f5f28fc88804076f41bb.

ty4g1 committed:
Add javadocs to new methods

ty4g1 committed:
Merge branch 'branch-Level-7'

ty4g1 committed:
Update .gitignore file to ignore stored data

ty4g1 committed:
Remove data files

ty4g1 committed:
Implement date formatting and relevant exceptions

ty4g1 committed:
Merge branch 'branch-Level-8'

ty4g1 committed:
Extract UI into its own class

ty4g1 committed:
Extract task list and storage into their own classes

ty4g1 committed:
Extract parsing and commands into their own classes

ty4g1 committed:
Update test case

ty4g1 committed:
Merge branch 'branch-A-MoreOOP'

ty4g1 committed:
Organize classes into packages

ty4g1 committed:
Modify Mahesh.java and add it to mahesh package

ty4g1 committed:
Fix incorrect path bugs in Storage

ty4g1 committed:
Merge branch 'branch-A-Packages'

ty4g1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ty4g1 committed:
Modify build.gradle file

ty4g1 committed:
Fix minor printing bug

ty4g1 committed:
Add JUnit test files

ty4g1 committed:
Merge branch 'branch-A-JUnit'

ty4g1 committed:
Change filename for Jar file in build.gradle

ty4g1 committed:
Merge branch 'branch-A-jar'

ty4g1 committed:
Implement find functionality

ty4g1 committed:
Merge branch 'branch-Level-9'

ty4g1 committed:
Add JavaDoc for files in mahesh.command package

ty4g1 committed:
Add JavaDoc for files in mahesh.util package

ty4g1 committed:
Add JavaDoc for Mahesh.java

ty4g1 committed:
Merge branch 'branch-A-JavaDoc'

ty4g1 committed:
Make constants in classes final

ty4g1 committed:
Merge branch 'branch-A-CodingStandard'

Oceankoh commented on src/test/java/mahesh/task/DeadlineTest.java:
```diff
@@ -0,0 +1,47 @@
+        }
+    }
+
+    @Test void testParseDeadlineWithIncompleteInput() {
```
Should the annotation be on a new line to be consistent with the rest of the tests annotations?

ty4g1 committed:
Add checkstyle config

ty4g1 committed:
Fix style errors in command package

ty4g1 committed:
Fix style errors in Mahesh and task package

ty4g1 committed:
Fix style errors in util package

ty4g1 committed:
Fix style errors in test files

ty4g1 committed:
Merge branch 'branch-A-CheckStyle'

ty4g1 committed:
Minor style changes

ty4g1 committed:
Remove trailing whitespaces

warg-void commented on src/main/java/mahesh/Mahesh.java:
Is calling store.updateData redundant in the command classes redundant because it is already called here?

warg-void reviewed (commented):
LGTM, I like that you've split the command classes neatly.

Oceankoh commented on src/main/java/mahesh/Mahesh.java:
```diff
@@ -0,0 +1,72 @@
+     * @param args command-line arguments (not used)
+     */
+    public static void main(String[] args) {
+        Mahesh app = new Mahesh("../../../../data/mahesh.txt");
```
Could using a string to store the path pose compatibility issues with other operating systems such as Windows?

Oceankoh commented on src/main/java/mahesh/util/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+            Ui.printEmptyListErr();
+            return;
+        }
+        StringBuilder result = new StringBuilder();
```
Small thing, but great use of StringBuilder to improve time complexity even though it might have negligible performance improvements :)

Oceankoh commented on src/main/java/mahesh/util/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+     * Prints all tasks in the list. Prints an error message if the list is empty.
+     */
+    public void printList() {
+        if (this.list.size() == 0) {
```
Maybe you could consider using the built-in `isEmpty()` method for this check?

Oceankoh commented on src/main/java/mahesh/util/Ui.java:
```diff
@@ -0,0 +1,147 @@
+     */
+    public static void printCorruptedDataErr() {
+        System.out.println("Some tasks may not have loaded correctly due to corrupted data file.");
+        System.out.println(DIVIDER);
```
Since you've defined `printDivider()`, maybe you could consider using in your other methods. This could be useful if you ever chose to add another print statement to your divider!

blackpanther9229 commented on src/main/java/mahesh/command/CommandNames.java:
```diff
@@ -0,0 +1,26 @@
+/**
+ * Enum representing the various commands that can be issued to the Mahesh bot.
+ */
+public enum CommandNames {
```
I recommend a more straighfoward name, which is `CommandEnum`

Oceankoh commented on src/test/java/mahesh/util/ParserTest.java:
```diff
@@ -0,0 +1,87 @@
+    @BeforeEach
+    public void setUp() {
+        taskList = new TaskList();
+        storage = new Storage("test.txt"); // Assuming Storage takes a file path
```
Is there a better way than creating a test file? (hint: stubs and mocks!)

Oceankoh reviewed (commented):
Nicely done!

blackpanther9229 commented on src/main/java/mahesh/task/Event.java:
```diff
@@ -0,0 +1,115 @@
+     * @return A string representation of the Event task for file storage.
+     */
+    @Override
+    public String toFileEntry() {
```
`toFileFormat` would be a better method name

ty4g1 committed:
Implement GUI

ty4g1 committed:
Update JavaDocs

ty4g1 committed:
Fix style issues

ty4g1 committed:
Merge branch 'branch-Level-10'

ty4g1 committed:
Modify overloaded constructors to use varargs instead

ty4g1 committed:
Revert "Modify overloaded constructors to use varargs instead"

This reverts commit 50ae00bc52df267256f05c5de2ff9bb9b6c9b52e.

ty4g1 committed:
Update JavaDocs

ty4g1 committed:
Merge branch 'branch-A-Varargs'

ty4g1 committed:
Add assertions in Mahesh and Parser

ty4g1 committed:
Fix long methods

ty4g1 committed:
Merge branch 'branch-A-Assertions'

ty4g1 committed:
Add assertions to TaskList

ty4g1 committed:
Merge pull request #1 from ty4g1/branch-A-Assertions

Add assertions to Mahesh, Parser and TaskList

ty4g1 committed:
Merge branch 'master' of github.com:ty4g1/ip

ty4g1 committed:
Merge branch 'master' into branch-A-CodeQuality

ty4g1 committed:
Merge branch 'master' of github.com:ty4g1/ip into branch-A-CodeQuality

ty4g1 committed:
Merge branch 'branch-A-CodeQuality' of github.com:ty4g1/ip into branch-A-CodeQuality

ty4g1 committed:
Merge pull request #2 from ty4g1/branch-A-CodeQuality

Fix long methods in Mahesh and Parser

ty4g1 committed:
Merge branch 'master' of github.com:ty4g1/ip

ty4g1 committed:
Add Recurring Tasks Functionality and Tests

- Implemented the `Recurring` class to represent recurring tasks.
  - Added constructors to initialize recurring tasks with description, day, and completion status.
  - Overrode `toString` method to include the recurring day in the string representation.
  - Overrode `toFileEntry` method to format the recurring task for file storage.
  - Added `parseRecurring` method to parse recurring tasks from a `StringTokenizer`.

- Updated the `Parser` class to handle recurring tasks.
  - Added a new case for the `RECURRING` command in the `parse` method.
  - Created a `parseRecurringCommand` method to handle the parsing of recurring tasks.

- Added unit tests for the `Recurring` class.
  - `testParseRecurring`: Tests the parsing of valid recurring tasks.
  - `testParseRecurringWithInvalidInput`: Tests the parsing of invalid recurring tasks.

- Ensured that the application correctly handles recurring tasks and passes all tests.

ty4g1 committed:
Merge pull request #3 from ty4g1/branch-B-RecurringTasks

Add Recurring Tasks Functionality and Tests

ty4g1 committed:
Update images and modify MainWindow.java

- Modified MainWindow.java to improve UI functionality.
- Deleted outdated image DaDuke.png.
- Added new image DaMahesh.png.
- Updated image DaUser.png to the latest version.

ty4g1 committed:
Update Ui.java to append "Mamta dum dum" to responses

- Modified printTaskAdded to append "Mamta dum dum" to the response.
- Modified printTaskDeleted to append "Mamta dum dum" to the response.
- Modified printMarkedAsDone to append "Mamta dum dum" to the response.
- Modified printUnmarkedAsDone to append "Mamta dum dum" to the
  response.
- Modified printErr to append "Mamta dum dum" to the response.
- Modified printNoSuchTaskErr to append "Mamta dum dum" to the response.
- Modified printIncompleteCommandErr to append "Mamta dum dum" to the
  response.
- Modified printEmptyListErr to append "Mamta dum dum" to the response.
- Modified printCorruptedDataErr to append "Mamta dum dum" to the
  response.

ty4g1 committed:
Add Ui.png to documentation

- Added new image file Ui.png to the docs directory for better
  visualization.

ty4g1 committed:
Update README.md with user guide and product intro

- Added detailed user guide for Mahesh Dall-E.
- Included product introduction to provide an overview of the tool.

ty4g1 committed:
Fix issue with jar file not including dependencies

ty4g1 committed:
Address issues from peer evaluation

ty4g1 committed:
Merge pull request #5 from ty4g1/fix-ip-issues

Address issues from peer evaluation

ty4g1 committed:
Update README

ty4g1 committed:
Add Ui image to UG

ty4g1 committed:
Fix minor issue with personality

ty4g1 committed:
Merge branch 'master' of github.com:ty4g1/ip
